kratos/includes/node.h

```diff
@@ -757,42 +757,60 @@ class Node : public Point,  public IndexedObject, public Flags
         typename DofsContainerType::iterator it_begin = mDofs.begin();
         typename DofsContainerType::iterator it_end = mDofs.end();
         typename DofsContainerType::iterator it;
-        //if the guess is exact return the guess
-        if(pos < it_end-it_begin)
-        {
+        // If the guess is exact return the guess
+        if(pos < it_end-it_begin) {
             it = it_begin + pos;
-            if( (it)->GetVariable() == rDofVariable)
-            {
+            if( (it)->GetVariable() == rDofVariable) {
                 return *it;
             }
         }
 
         // Otherwise do a find
-        it = mDofs.find(rDofVariable.Key());
-        if ( it!= mDofs.end() )
-        {
-            return *it;
+        return GetDof(rDofVariable);
+    }
+
+    template<class TVariableType>
+    inline DofType& GetDof(TVariableType const& rDofVariable, int pos) const
+    {
+        typename DofsContainerType::iterator it_begin = mDofs.begin();
+        typename DofsContainerType::iterator it_end = mDofs.end();
+        typename DofsContainerType::iterator it;
+        // If the guess is exact return the guess
+        if(pos < it_end-it_begin) {
+            it = it_begin + pos;
+            if( (it)->GetVariable() == rDofVariable) {
+                return *it;
+            }
         }
 
-        std::stringstream buffer;
-        buffer << "Not existant DOF in node #" << Id() << " for variable : " << rDofVariable.Name();
-        KRATOS_ERROR <<  buffer.str() << std::endl;
+        // Otherwise do a find
+        return GetDof(rDofVariable);
     }
 
     /** returns the Dof asociated with variable  */
     template<class TVariableType>
     inline DofType& GetDof(TVariableType const& rDofVariable)
     {
         typename DofsContainerType::iterator it=mDofs.find(rDofVariable.Key());
-        if ( it!= mDofs.end() )
-        {
+        if ( it!= mDofs.end() ) {
             return *it;
         }
 
-        std::stringstream buffer;
-        buffer << "Not existant DOF in node #" << Id() << " for variable : " << rDofVariable.Name();
-        KRATOS_ERROR <<  buffer.str() << std::endl;
+        KRATOS_ERROR <<  "Not existant DOF in node #" << Id() << " for variable : "

```

Comment:
just wondering if this should not only be thrown in debug for efficiency reasons (i appreciate it would be a behaviour change and does not belong in this PR)

Reply:
If it is not an error it would not arrive to this statement. So I don't see it as a performance problem